File: src/main/java/com/example/gagso/Documents/dto/DocumentDTO.java
package com.example.gagso.Documents.dto;

import com.example.gagso.Documents.models.Attachment;
import jakarta.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.beans.Visibility;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DocumentDTO {
    private String docID;
    private String writerID;
    private String title;
    private String content;
    private String visibility;
    private LocalDate date;
    private List<AttachmentDTO> attachments;
}

File: src/main/frontend/src/Pages/documents/DocumentDTO.ts
export interface DocumentDTO {
  docID: string;
  writerID: string;
  content: string;
  title: string;
  visibility: string;
  date: Date;
  attachments: AttachmentDTO[];
}

export interface AttachmentDTO {
  id: number;
  originalName: string;
  path: string;
  size: number;
}
